### PR TITLE
Rewrote DB and finished LoginServer

### PR DESCRIPTION
**Pull request by kamilprzyb**
- DB now uses ORM
- Fields of Player Characters are stored properly, instead of Json
- LoginServer checks if account isn't already logged in, and let's you kill the other connection
- Name approval system is implemented (but can't be called yet) , with a new setting APPROVEALLNAMES (true by default)
- a lot of refactoring

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,11 +24,8 @@ CSRC=\
 	src/contrib/bcrypt/crypt_blowfish.c\
 	src/contrib/bcrypt/crypt_gensalt.c\
 	src/contrib/bcrypt/wrapper.c\
-	src/contrib/sqlite/sqlite3.c\
```
Comment (FinnHornhoover) [inline Makefile:27]: This and the deletion of `src/contrib/sqlite/sqlite3*.h` cause problems for the Linux builds.

File: src/Database.cpp [hunk 2]
```diff
@@ -1,424 +1,356 @@
...
-        DEBUGLOG(std::cout << "Done" << std::endl;)
-    }
+#include "CNStructs.hpp"
+#include "contrib/sqlite/sqlite_orm.h"
+
+using namespace sqlite_orm;
+
+# pragma region DatabaseScheme
+auto db = make_storage("database.db",
+	make_table("Accounts",
```
Comment (FinnHornhoover) [inline src/Database.cpp:16]: Looks like the file was indented with tabs, can you change that to 4 spaces per tab instead?

File: src/Database.hpp [hunk 3]
```diff
@@ -1,31 +1,91 @@
...
 #include "CNStructs.hpp"
 #include "Player.hpp"
-#include <list>
 #include <string>
+#include <vector>
 
 namespace Database {
+#pragma region DatabaseStructs
+
+	struct Account
```
Comment (FinnHornhoover) [inline src/Database.hpp:10]: Tab indentation here as well, let's fix that

File: src/settings.cpp [hunk 4]
```diff
@@ -35,6 +36,7 @@ void settings::init() {
         return;
     }
 
+    APPROVEALLNAMES = reader.GetInteger("", "acceptallcustomnames", APPROVEALLNAMES);
```
Comment (FinnHornhoover) [inline src/settings.cpp:39]:
```suggestion
    APPROVEALLNAMES = reader.GetBoolean("", "acceptallcustomnames", APPROVEALLNAMES);
```

File: src/Database.cpp (hunk 2 above)
Comment (kamilprzyb) [reply src/Database.cpp]: fixed, and changed my editor settings for the future

File: src/Database.hpp (hunk 3 above)
Comment (kamilprzyb) [reply src/Database.hpp]: fixed, and changed my editor settings for the future

File: Makefile (hunk 1 above)
Comment (kamilprzyb) [reply Makefile:27]: fixed

**Pull request merged**